src/vma/util/sysctl_reader.h

```diff
@@ -140,6 +146,20 @@ public :
 		return &tcp_mem;
 	}
 
+	const tcp_keepalive_info *get_tcp_keepalive_info(bool update = false)
+	{
+		static tcp_keepalive_info info = {7200, 75, 9};
+		if (update) {
+			info.idle_secs =
+				read_file_to_int("/proc/sys/net/ipv4/tcp_keepalive_time", info.idle_secs);
+			info.interval_secs =
+				read_file_to_int("/proc/sys/net/ipv4/tcp_keepalive_intvl", info.interval_secs);
+			info.num_probes =
+				read_file_to_int("/proc/sys/net/ipv4/tcp_keepalive_probes", info.num_probes);
+		}
+		return &info;
+	}
+
```

Comment:
Please consider following
1. return tcp_keepalive_info instead of tcp_keepalive_info*
2. use `val` variable name instead of `info` to align with other methods.

Reply:
> 2. use val variable name instead of info to align with other methods.
NP

> 1. return tcp_keepalive_info instead of tcp_keepalive_info*
I guess the argument for this is consistency with the other functions in the file, but contrary to the other functions, this one returns a struct. Moreover, the struct is stored in a static variable, so we are guaranteed to copy it. I believe this is unnecessary. @igor-ivanov , don't you agree?

Reply:
so there is no issue with returning struct, right?

Reply:
OK